[One Discover] [Logs UX]Add support to disable highlighting for certain filters we deem internal

achyutjhunjhunwala opened the pull request:
## Summary

closes https://github.com/elastic/kibana/issues/234215

Implements selective filter highlighting to prevent context/internal filters from being highlighted in search results while still using them for document retrieval.

## Problem

When using SavedSearchComponent with filters (e.g., filtering logs to `service.name: synth-node-0`), all filter matches are highlighted in yellow. This creates visual noise when filters are used for context rather than active search criteria.

For example, in APM Service Logs:
- Service name filter is used to scope the view but shouldn't be highlighted
- User search filters should be highlighted to help locate matches

Previously, there was no way to distinguish between these two types of filters.

## Solution

This PR uses Elasticsearch's `highlight_query` parameter to control which filters trigger highlighting. The implementation adds a `skipHighlight` metadata field to filters.

- Consumers mark filters with `skipHighlight: true` when creating them:

- SearchSource builds two queries:

1. **Main query** - Uses all filters for document retrieval
2. **Highlight query** - Excludes filters with `skipHighlight: true`

Example ES request:
```json
{
  "query": {
    "bool": {
      "filter": [
        { "term": { "service.name": "synth.node" } },
        { "term": { "host.name": "instance" } }
      ]
    }
  },
  "highlight": {
    "fields": { "*": {} },
    "highlight_query": {
      "bool": {
        "filter": [
          { "term": { "host.name": "instance" } }
        ]
      }
    }
  }
}
```

Result: Documents are filtered by both conditions, but only "instance" is highlighted.

## Performance Impact

The highlight_query parameter is processed by Elasticsearch on already-retrieved documents, so there's no additional data fetching or query execution overhead.

## Screenshots

### Before (APM)

<img width="2280" height="1061" alt="image" src="https://github.com/user-attachments/assets/9a1cb62a-2252-443e-9c9c-46c49d301edc" />

### After

<img width="2294" height="1087" alt="image" src="https://github.com/user-attachments/assets/c800ae3c-3b4c-4b7d-bf09-04260d90db81" />

### Before (Host View)

<img width="1924" height="1055" alt="image" src="https://github.com/user-attachments/assets/7b758ffd-dd25-4a1b-8895-9f4941a671d5" />


### After

<img width="1958" height="1130" alt="image" src="https://github.com/user-attachments/assets/9aeff451-95ea-4f19-9177-60068dfd7aee" />

### After (Hosts Logs inside Flyout)

<img width="2296" height="1172" alt="image" src="https://github.com/user-attachments/assets/83a28a93-8dec-4708-b717-1c4966f93834" />


### Before (Discover flyout for traces)

<img width="1578" height="1138" alt="image" src="https://github.com/user-attachments/assets/9205d2bc-aeb7-4eda-af0a-35377862b45d" />


### After

<img width="1580" height="1175" alt="image" src="https://github.com/user-attachments/assets/063b4726-2a79-4cad-9755-0c51bacf234b" />


## Logic Implemented for ?

- APM Logs Page
- All Host View
- Host View inside Flyout
- Discover Flyout for Traces

achyutjhunjhunwala committed:
Add support to disable highlighting for certain filters we deem internal

achyutjhunjhunwala committed:
Replicate behaviour to Hosts view

achyutjhunjhunwala committed:
Fix logic in host view to filter only user specific filters

achyutjhunjhunwala committed:
Merge branch 'main' into add-support-for-hidden-filters-in-logs-overview-component

achyutjhunjhunwala committed:
Add logic to control highlighting

elasticmachine commented:
Pinging @elastic/obs-ux-logs-team (Team:obs-ux-logs)

dej611 commented on src/platform/plugins/shared/data/common/search/search_source/search_source.ts:
```diff
@@ -879,21 +879,40 @@ export class SearchSource {
+    const allFilters =
+      typeof searchRequest.filters === 'function'
+        ? searchRequest.filters()
+        : searchRequest.filters || [];
```
```suggestion
        : searchRequest.filters ?? [];
```

dej611 reviewed (approved):
Code review only. Left minor nit comment.

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

achyutjhunjhunwala committed:
Update src/platform/plugins/shared/data/common/search/search_source/search_source.ts

Co-authored-by: Marco Liberati <dej611@users.noreply.github.com>

jennypavlova reviewed (changes requested):
Thanks for the highlighting changes! I tested on the host view, and if the user makes a typo, it crashes the whole page (The refresh button is not helping, and nothing works except leaving the app 

https://github.com/user-attachments/assets/0e079fa6-a391-4a69-a146-4455a0f747a2


<img width="1810" height="968" alt="error" src="https://github.com/user-attachments/assets/8ff24b5b-6403-404d-9bc7-8f85b695fb70" />

In this case, I would expect a validation around the search field or similar to what we have in the metadata tab.
<img width="903" height="436" alt="image" src="https://github.com/user-attachments/assets/9c689ec8-9ea7-4954-8ed4-53d5b992df3e" />

Can you please take a look?

Also, on the single host view, the page is not crashing, but I can only search using a string `log. level: "info"` won't work there (and if the string is "Error," for example, I see mixhighlighting - which I guess is expected) 


https://github.com/user-attachments/assets/ec065b8b-da08-4cc0-b4c6-2d341d8c540b

crespocarlos commented on x-pack/solutions/observability/plugins/infra/public/pages/metrics/hosts/components/tabs/logs/logs_tab_content.tsx:
```diff
@@ -34,19 +34,57 @@ export const LogsTabContent = () => {
+  const logsSearchFilters = useMemo(
+    () =>
+      filterQuery && filterQuery.query
+        ? [buildEsQuery(undefined, filterQuery, [], getEsQueryConfig(uiSettings))]
```
we probably need an error handling here too

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/353008)
* Commit: 5550b9ef87dbbc27e6a2b60e98d84a4318b8cec0

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b5314276e7299f16b170d91b20ec5e07a52a3160) | [after](https://github.com/elastic/kibana/commit/5550b9ef87dbbc27e6a2b60e98d84a4318b8cec0) | diff |
| --- | --- | --- | --- |
| `apm` | 2.8MB | 2.8MB | +64.0B |
| `infra` | 1.1MB | 1.1MB | +1004.0B |
| `logsShared` | 203.1KB | 203.5KB | +414.0B |
| total |  |  | +1.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/b5314276e7299f16b170d91b20ec5e07a52a3160) | [after](https://github.com/elastic/kibana/commit/5550b9ef87dbbc27e6a2b60e98d84a4318b8cec0) | diff |
| --- | --- | --- | --- |
| `data` | 434.2KB | 434.7KB | +549.0B |
| `kbnUiSharedDeps-srcJs` | 3.9MB | 3.9MB | +18.0B |
| total |  |  | +567.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/b5314276e7299f16b170d91b20ec5e07a52a3160) | [after](https://github.com/elastic/kibana/commit/5550b9ef87dbbc27e6a2b60e98d84a4318b8cec0) | diff |
| --- | --- | --- | --- |
| `@kbn/es-query` | 303 | 304 | +1 |


</details>


### History
* :yellow_heart: [Build #352590](https://buildkite.com/elastic/kibana-pull-request/builds/352590) was flaky 11f9cccbdfb15cc63d93830ed20d192ca4f6d310
* :yellow_heart: [Build #352242](https://buildkite.com/elastic/kibana-pull-request/builds/352242) was flaky 8eeedc6242c4aa8206a3c39baf9ee690fadbbde2
* :yellow_heart: [Build #352053](https://buildkite.com/elastic/kibana-pull-request/builds/352053) was flaky bef912cea33d1196f830e75036fe3f9d32d450d9

cc @achyutjhunjhunwala

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353008","number":353008,"commit":"5550b9ef87dbbc27e6a2b60e98d84a4318b8cec0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352590","number":352590,"commit":"11f9cccbdfb15cc63d93830ed20d192ca4f6d310"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352242","number":352242,"commit":"8eeedc6242c4aa8206a3c39baf9ee690fadbbde2"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352053","number":352053,"commit":"bef912cea33d1196f830e75036fe3f9d32d450d9"}],"number":353008}
buildkite-pr-comment-->

achyutjhunjhunwala committed:
Fix review comments

achyutjhunjhunwala committed:
apply the same code review to service logs as well

crespocarlos reviewed (commented):
LGTM. CR only. Thanks for the adjustments

achyutjhunjhunwala committed:
Merge branch 'main' into add-support-for-hidden-filters-in-logs-overview-component

jennypavlova reviewed (approved):
LGTM 💯 Thanks for fixing the hosts view error!

jennypavlova commented:
BTW, if I search for "Error", I see 2 different highlighting colors (more of a visual nit, could be checked with Design / Product what should be the correct color in this case, or if it should stay like that) 

<img width="1572" height="859" alt="image" src="https://github.com/user-attachments/assets/820683b5-5e6c-4f0a-854e-04fba322da21" />

achyutjhunjhunwala committed:
Merge branch 'main' into add-support-for-hidden-filters-in-logs-overview-component

rStelmach reviewed (approved):
LGTM!

lukasolson reviewed (commented):
I'm not a huge fan of including the `skipHighlight` in the filter meta, simply because this is more of a UX representation of the filter, and there's no way in the UX to configure `skipHighlight`. This means you could get into an odd situation where you have filters showing up and they're not highlighting as expected, and there's no indication how to fix this.

I did notice @davismcphee suggested [here](https://github.com/elastic/kibana/issues/234215#issuecomment-3309660873) trying to use component props, and I'm curious if that approach was attempted, and if so, what was challenging or didn't work as expected. I think what would make more sense to me is if we added an additional `highlightQuery` prop to the `SavedSearchComponent` which allowed passing in DSL for the highlight query (or even the entire `highlight` configuration) and then used something like `searchSource.setField('highlight', highlightProps)` in the same place we're setting `query` and `filters`.

achyutjhunjhunwala committed:
Merge branch 'main' into add-support-for-hidden-filters-in-logs-overview-component

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/357659)
* Commit: 451d4d7394e819db2af4385f331d3cdfb8853b14

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/86d6d12a3890fda28d105ff309fbd418482a52b3) | [after](https://github.com/elastic/kibana/commit/451d4d7394e819db2af4385f331d3cdfb8853b14) | diff |
| --- | --- | --- | --- |
| `apm` | 2.8MB | 2.8MB | +58.0B |
| `infra` | 1.1MB | 1.1MB | +1.1KB |
| `logsShared` | 198.2KB | 198.7KB | +414.0B |
| total |  |  | +1.6KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/86d6d12a3890fda28d105ff309fbd418482a52b3) | [after](https://github.com/elastic/kibana/commit/451d4d7394e819db2af4385f331d3cdfb8853b14) | diff |
| --- | --- | --- | --- |
| `data` | 433.2KB | 433.8KB | +605.0B |
| `kbnUiSharedDeps-srcJs` | 4.0MB | 4.0MB | +18.0B |
| total |  |  | +623.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/86d6d12a3890fda28d105ff309fbd418482a52b3) | [after](https://github.com/elastic/kibana/commit/451d4d7394e819db2af4385f331d3cdfb8853b14) | diff |
| --- | --- | --- | --- |
| `@kbn/es-query` | 303 | 304 | +1 |


</details>


### History
* :yellow_heart: [Build #353642](https://buildkite.com/elastic/kibana-pull-request/builds/353642) was flaky adea8efcd37f3e92b6eee00546d18278077a737d
* :yellow_heart: [Build #353537](https://buildkite.com/elastic/kibana-pull-request/builds/353537) was flaky 77d2b7eb3421567bba566e5f40d85f7984671d13
* :broken_heart: [Build #353183](https://buildkite.com/elastic/kibana-pull-request/builds/353183) failed eefddd30dc12f447def2301d9fabb7fd03db73c2
* :green_heart: [Build #353008](https://buildkite.com/elastic/kibana-pull-request/builds/353008) succeeded 5550b9ef87dbbc27e6a2b60e98d84a4318b8cec0
* :yellow_heart: [Build #352590](https://buildkite.com/elastic/kibana-pull-request/builds/352590) was flaky 11f9cccbdfb15cc63d93830ed20d192ca4f6d310
* :yellow_heart: [Build #352242](https://buildkite.com/elastic/kibana-pull-request/builds/352242) was flaky 8eeedc6242c4aa8206a3c39baf9ee690fadbbde2

cc @achyutjhunjhunwala

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357659","number":357659,"commit":"451d4d7394e819db2af4385f331d3cdfb8853b14"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353642","number":353642,"commit":"adea8efcd37f3e92b6eee00546d18278077a737d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353537","number":353537,"commit":"77d2b7eb3421567bba566e5f40d85f7984671d13"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353183","number":353183,"commit":"eefddd30dc12f447def2301d9fabb7fd03db73c2"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353008","number":353008,"commit":"5550b9ef87dbbc27e6a2b60e98d84a4318b8cec0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352590","number":352590,"commit":"11f9cccbdfb15cc63d93830ed20d192ca4f6d310"}],"number":357659}
buildkite-pr-comment-->

achyutjhunjhunwala committed:
Remove skip highlight meta filter and replace the logic with passing props

achyutjhunjhunwala committed:
Fix some of the old review comments

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

achyutjhunjhunwala committed:
Add skip highlight for Discover Traces Flyout which shows logs

achyutjhunjhunwala committed:
Merge branch 'main' into add-support-for-hidden-filters-in-logs-overview-component

achyutjhunjhunwala commented:
@iblancof I have added the support for the Traces Flyout in Discover, check the last commit [here](https://github.com/elastic/kibana/pull/239402/commits/2d209e3f288023abc2cce9032bb337c441afa6dd). Also updated the description with a screenshot

@lukasolson I have changed the logic and removed the `skipHighlight` meta filter and moved the logic to props. 

PR is ready for review again

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/358094)
* Commit: ecfa6d7525427c17dbc4066a54bc748b29107bc4

### Failed CI Steps
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/358094#019a4ebb-240e-4947-b6c1-9107d50bad43)
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/358094#019a4ece-c98f-467d-89bd-e9d229aa28d0)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358094#019a4ebb-240e-4947-b6c1-9107d50bad43) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358094/jobs/019a4ebb-240e-4947-b6c1-9107d50bad43/artifacts/019a4ece-a957-4d1e-b737-b5a34076cbbd) Jest Tests #<span></span>11 / LogsOverview with a data view as logs source displays the log events from the passed index name
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358094#019a4ece-c98f-467d-89bd-e9d229aa28d0) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358094/jobs/019a4ece-c98f-467d-89bd-e9d229aa28d0/artifacts/019a4ee7-3587-43cc-8913-2218c4b87206) Jest Tests #<span></span>11 / LogsOverview with a data view as logs source displays the log events from the passed index name
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358094#019a4ebb-240e-4947-b6c1-9107d50bad43) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358094/jobs/019a4ebb-240e-4947-b6c1-9107d50bad43/artifacts/019a4ece-a956-4397-b6e6-746a61c757bc) Jest Tests #<span></span>11 / LogsOverview with an index name as logs source displays the log events from the passed index name
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358094#019a4ece-c98f-467d-89bd-e9d229aa28d0) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358094/jobs/019a4ece-c98f-467d-89bd-e9d229aa28d0/artifacts/019a4ee7-3586-4e6e-90d3-75c4a4ad2fe9) Jest Tests #<span></span>11 / LogsOverview with an index name as logs source displays the log events from the passed index name
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358094#019a4ebb-240e-4947-b6c1-9107d50bad43) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358094/jobs/019a4ebb-240e-4947-b6c1-9107d50bad43/artifacts/019a4ece-a957-4999-a770-fa6d4ad99c0b) Jest Tests #<span></span>11 / LogsOverview without the default logs source displays the log events from the shared settings by default
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358094#019a4ece-c98f-467d-89bd-e9d229aa28d0) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358094/jobs/019a4ece-c98f-467d-89bd-e9d229aa28d0/artifacts/019a4ee7-3586-4206-b0cc-28f602d509e7) Jest Tests #<span></span>11 / LogsOverview without the default logs source displays the log events from the shared settings by default
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358094#019a4ece-c98f-467d-89bd-e9d229aa28d0) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358094/jobs/019a4ece-c98f-467d-89bd-e9d229aa28d0/artifacts/019a4ee7-3586-4345-950a-9d7c0e5c22a5) Jest Tests #<span></span>11 / UpdateConnector renders should confirm the update when submit button clicked

### History
* :green_heart: [Build #357659](https://buildkite.com/elastic/kibana-pull-request/builds/357659) succeeded 451d4d7394e819db2af4385f331d3cdfb8853b14
* :yellow_heart: [Build #353642](https://buildkite.com/elastic/kibana-pull-request/builds/353642) was flaky adea8efcd37f3e92b6eee00546d18278077a737d
* :yellow_heart: [Build #353537](https://buildkite.com/elastic/kibana-pull-request/builds/353537) was flaky 77d2b7eb3421567bba566e5f40d85f7984671d13
* :broken_heart: [Build #353183](https://buildkite.com/elastic/kibana-pull-request/builds/353183) failed eefddd30dc12f447def2301d9fabb7fd03db73c2
* :green_heart: [Build #353008](https://buildkite.com/elastic/kibana-pull-request/builds/353008) succeeded 5550b9ef87dbbc27e6a2b60e98d84a4318b8cec0
* :yellow_heart: [Build #352590](https://buildkite.com/elastic/kibana-pull-request/builds/352590) was flaky 11f9cccbdfb15cc63d93830ed20d192ca4f6d310

cc @achyutjhunjhunwala

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358094","number":358094,"commit":"ecfa6d7525427c17dbc4066a54bc748b29107bc4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357659","number":357659,"commit":"451d4d7394e819db2af4385f331d3cdfb8853b14"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353642","number":353642,"commit":"adea8efcd37f3e92b6eee00546d18278077a737d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353537","number":353537,"commit":"77d2b7eb3421567bba566e5f40d85f7984671d13"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353183","number":353183,"commit":"eefddd30dc12f447def2301d9fabb7fd03db73c2"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353008","number":353008,"commit":"5550b9ef87dbbc27e6a2b60e98d84a4318b8cec0"}],"number":358094}
buildkite-pr-comment-->

lukasolson commented on src/platform/plugins/shared/data/common/search/search_source/types.ts:
```diff
@@ -80,6 +80,13 @@ export interface SearchSourceFields {
+   * but excluded from the highlight_query parameter in Elasticsearch.
+   * {@link Filter}
+   */
+  nonHighlightingFilters?: Filter[] | Filter | (() => Filter[] | Filter | undefined);
```
If possible, I'd like to avoid copying this tech debt into the new field:

```suggestion
  nonHighlightingFilters?: Filter[];
```

lukasolson reviewed (commented):
Thanks for making these changes, this feels better! I just left some minor comments below about not typing `nonHighlightingFilters` as a function.

achyutjhunjhunwala committed:
Fix more review comments and broken test snapshots

achyutjhunjhunwala committed:
Merge branch 'main' into add-support-for-hidden-filters-in-logs-overview-component

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/358479)
* Commit: 7f5cb01bda6e67fe3c7495dd44083ca60f33d477

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/131f1a002666e1a3065dacc30754a519bce7a2cc) | [after](https://github.com/elastic/kibana/commit/7f5cb01bda6e67fe3c7495dd44083ca60f33d477) | diff |
| --- | --- | --- | --- |
| `apm` | 2.8MB | 2.8MB | +124.0B |
| `fleet` | 2.1MB | 2.1MB | +66.0B |
| `infra` | 1.1MB | 1.1MB | +1.2KB |
| `logsShared` | 198.2KB | 198.8KB | +529.0B |
| `unifiedDocViewer` | 274.0KB | 274.0KB | +15.0B |
| total |  |  | +1.9KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/131f1a002666e1a3065dacc30754a519bce7a2cc) | [after](https://github.com/elastic/kibana/commit/7f5cb01bda6e67fe3c7495dd44083ca60f33d477) | diff |
| --- | --- | --- | --- |
| `data` | 433.2KB | 433.9KB | +678.0B |
| `logsShared` | 90.0KB | 90.6KB | +585.0B |
| total |  |  | +1.2KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/131f1a002666e1a3065dacc30754a519bce7a2cc) | [after](https://github.com/elastic/kibana/commit/7f5cb01bda6e67fe3c7495dd44083ca60f33d477) | diff |
| --- | --- | --- | --- |
| `@kbn/saved-search-component` | 16 | 17 | +1 |
| `data` | 3208 | 3210 | +2 |
| total |  |  | +3 |


</details>


### History
* :broken_heart: [Build #358094](https://buildkite.com/elastic/kibana-pull-request/builds/358094) failed ecfa6d7525427c17dbc4066a54bc748b29107bc4
* :green_heart: [Build #357659](https://buildkite.com/elastic/kibana-pull-request/builds/357659) succeeded 451d4d7394e819db2af4385f331d3cdfb8853b14
* :yellow_heart: [Build #353642](https://buildkite.com/elastic/kibana-pull-request/builds/353642) was flaky adea8efcd37f3e92b6eee00546d18278077a737d
* :yellow_heart: [Build #353537](https://buildkite.com/elastic/kibana-pull-request/builds/353537) was flaky 77d2b7eb3421567bba566e5f40d85f7984671d13
* :broken_heart: [Build #353183](https://buildkite.com/elastic/kibana-pull-request/builds/353183) failed eefddd30dc12f447def2301d9fabb7fd03db73c2

cc @achyutjhunjhunwala

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358479","number":358479,"commit":"7f5cb01bda6e67fe3c7495dd44083ca60f33d477"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358094","number":358094,"commit":"ecfa6d7525427c17dbc4066a54bc748b29107bc4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357659","number":357659,"commit":"451d4d7394e819db2af4385f331d3cdfb8853b14"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353642","number":353642,"commit":"adea8efcd37f3e92b6eee00546d18278077a737d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353537","number":353537,"commit":"77d2b7eb3421567bba566e5f40d85f7984671d13"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353183","number":353183,"commit":"eefddd30dc12f447def2301d9fabb7fd03db73c2"}],"number":358479}
buildkite-pr-comment-->
